File: Backend/Travle.Model/Requests/UserRegisterRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// Self-service registration. The client supplies only profile data — never a role or any
    /// privileged flag (course §J); the service always assigns the Traveler role.
    /// </summary>
    public class UserRegisterRequest
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; }
    }
}

File: UI/travle_core/lib/src/models/user_register_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_register_request.g.dart';

/// Self-service registration payload. The server always assigns the Traveler
/// role — the client never sends a role or privileged flag.
@JsonSerializable()
class UserRegisterRequest {
  UserRegisterRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.username,
    required this.password,
    this.phoneNumber,
  });

  final String firstName;
  final String lastName;
  final String email;
  final String username;
  final String password;
  final String? phoneNumber;

  factory UserRegisterRequest.fromJson(Map<String, dynamic> json) =>
      _$UserRegisterRequestFromJson(json);

  Map<String, dynamic> toJson() => _$UserRegisterRequestToJson(this);
}
